### PR TITLE
Lazy Tab ViewModels

### LINKED ISSUE #6679: Load and initialize the VMs for all tabs lazily
Currently when the user opens the app (cold start), the `MainTabBarViewModel` is loaded. This VM creates its child VMs in its constructor (main VM, reports VM, calendar VM) and the initialization method of the VM initializes all of the tab VMs.

I think we should change this:
- we should only create and initialize the main VM during startup
- the rest of the VMs should be either created and initialized in a background thread without blocking the UI
- or we should simply load the VMs and initialize them at the first occasion when the user navigates to the tabs (the user might not go to calendar or reports every time he/she opens the app)

Since this is all `Toggl.Core` territory, we must make sure we don't break either the iOS or the Android app while we're doing this. Both of these app's performance would benefit from this change.

### PR DESCRIPTION
**Pull request by heytherewill**
## What's this?
This makes the ViewModels used in tabs lazy

### Relationships
Closes #6679 
Brings the startup speed time from 1700ms to 1360ms (-340ms)

## Why do we want this?
No real reason to load everything up-front when you can wait

## How is it done?
Make things lazy on Droid. I'm not really making anything lazy on iOS, mostly because performance there is not heavily affected anyways.

### Side effects
The PR is as small as possible, but let's make sure we are not introducing bugs here, testing specially what happens when the app is collected on the BG

## :squid: Permissions
Me

### CODE REVIEW

**Comment by daividssilverio** [comment]
Btw, I made a quick experiment replacing the tabs in the main tab bar to start with the reports view and the cold startup itself wasn't affected... (In debug though).

**Comment by heytherewill** [comment]
@daividssilverio The problem is that the dependencies themselves aren't lazy enough. See https://github.com/toggl/mobileapp/issues/6977 and https://github.com/toggl/mobileapp/issues/7024

File: Toggl.iOS/ViewControllers/MainTabBarController.cs [hunk 1]
```diff
@@ -35,10 +37,16 @@ public sealed partial class MainTabBarController : UITabBarController
...
+                .GetAwaiter()
+                .GetResult();
 
-            UIViewController createTabFor(ViewModel childViewModel)
+            async Task<UIViewController> createTabFor(Lazy<ViewModel> lazViewModel)
             {
+                var childViewModel = lazViewModel.Value;
+                await childViewModel.Initialize();
                 var viewController = ViewControllerLocator.GetNavigationViewController(childViewModel);
                 var childViewModelType = childViewModel.GetType();
```
Comment (arturs105) [inline Toggl.iOS/ViewControllers/MainTabBarController.cs]: Don't think the `child` prefix of `childViewModel` is necessary anymore.

**Comment by heytherewill** [comment]
@semanticer Problem number 1 is fixed by @daividssilverio's proposal for #4423, so I'll let that be solved in the other issue :)

**Comment by semanticer** [comment]
but #2 is still not solved for me though @heytherewill

**Comment by daividssilverio** [comment]
Quick question, why did the `Toggl.sln` file changed?

**Comment by heytherewill** [comment]
@daividssilverio I rolled that back, thanks for catching it!

**Pull request merged**